packages/wrangler/src/index.tsx

```diff
@@ -100,6 +100,20 @@ function getScriptName(
     : shortScriptName;
 }
 
+async function requireAuth(config: Config): Promise<string> {
+  const loggedIn = await loginOrRefreshIfRequired();
+  if (!loggedIn) {
+    // didn't login, let's just quit
+    throw new Error("Did not login, quitting...");
```

Comment:
I would propose that we create a new error type, which we throw here and below.
This would enable two things:
- to render a more suitable message to the user (e.g. not saying "do you think this is a bug?")
- to avoid sending these "user errors" to Sentry.

Reply:
Oh and also we might be able to do something to fix this: https://github.com/cloudflare/wrangler2/issues/539

Reply:
I agree on this, but am torn whether to include it in this PR, thought it would make more sense with https://github.com/cloudflare/wrangler2/issues/515, https://github.com/cloudflare/wrangler2/issues/444. I wanted this one to be a simple refactor.

Reply:
👍  for doing that in another PR.